database/legacy/v3/migrate.go

```diff
@@ -190,6 +196,42 @@ func (db *Migrator) deleteOldMessagesByAddressFunction() error {
 	return err
 }
 
+func (db *Migrator) modifyOldMessageByAddressFunction() error {
+	_, err := db.Sql.Exec("DROP FUNCTION IF EXISTS messages_by_address(text[],text[],bigint,bigint);")
```

Comment:
cuz the function body will not be changed, the func will not use the _old tables :-/
```sql
# By simply altering the function name
bdjuno=> SELECT messages_by_address_old('{sif1rt5llnthdkvshew5a7qu4dwq9r5kv8tur9g3kz}','{}',2,0);

ERROR:  column reference "height" is ambiguous
LINE 7: ORDER BY height DESC
                 ^
QUERY:  
SELECT message.transaction_hash, message.index, message.type, message.value, message.involved_accounts_addresses
FROM message
         JOIN transaction t on message.transaction_hash = t.hash
WHERE (cardinality(types) = 0 OR type = ANY (types))
  AND addresses && involved_accounts_addresses
ORDER BY height DESC
LIMIT "limit" OFFSET "offset"
```


Otherwise, I think we can directly drop the old function and create the new one before migrating the rows, so the new function will be effective over new tables(though limited data before the migration is done). 

and since the new function is in place, the user can already restart the hasura docker container too to apply the function change while the rows are still being migrated. 

What do you think?

Reply:
Ok I see. We can modify old function to `messages_by_address_old` and create new `message_by_address` function before starting migration. This way the frontend can display transactions from the partitioned table without making any changes or it can also display transactions from `transactions_old` table during the migration process. After the migration is done we can drop `messages_by_address_old` function.